src/PowerShellEditorServices/Extensions/Api/WorkspaceService.cs

```diff
@@ -0,0 +1,160 @@
+        /// </summary>
+        public string Content { get; }
+
+        /// <summary>
+        /// The lines of the file.
+        /// </summary>
+        public IReadOnlyList<string> Lines { get; }
+
+        /// <summary>
+        /// The PowerShell AST of the script in the file.
+        /// </summary>
+        public ScriptBlockAst Ast { get; }
+
+        /// <summary>
+        /// The PowerShell syntactic tokens of the script in the file.
+        /// </summary>
+        public IReadOnlyList<Token> Tokens { get; }
+    }
+
+    /// <summary>
+    /// A service for querying and manipulating the editor workspace.
+    /// </summary>
+    public interface IWorkspaceService
+    {
+        /// <summary>
+        /// The root path of the workspace.
+        /// </summary>
+        string WorkspacePath { get; }
+
+        /// <summary>
+        /// Indicates whether the editor is configured to follow symlinks.
+        /// </summary>
+        bool FollowSymlinks { get; }
+
+        /// <summary>
+        /// The list of file globs to exclude from workspace management.
+        /// </summary>
+        IReadOnlyList<string> ExcludedFileGlobs { get; }
+
+        /// <summary>
+        /// Get a file within the workspace.
+        /// </summary>
+        /// <param name="fileUri">The absolute URI of the file to get.</param>
+        /// <returns>A representation of the file.</returns>
+        IEditorScriptFile GetFile(Uri fileUri);
+
+        /// <summary>
+        /// Attempt to get a file within the workspace.
+        /// </summary>
+        /// <param name="fileUri">The absolute URI of the file to get.</param>
+        /// <param name="file">The file, if it was found.</param>
+        /// <returns>True if the file was found, false otherwise.</returns>
+        bool TryGetFile(Uri fileUri, out IEditorScriptFile file);
+
+        /// <summary>
+        /// Get all the open files in the editor workspace.
+        /// The result is not kept up to date as files are opened or closed.
+        /// </summary>
+        /// <returns>All open files in the editor workspace.</returns>
+        IReadOnlyList<IEditorScriptFile> GetOpenedFiles();
+    }
+
+    internal class EditorScriptFile : IEditorScriptFile
+    {
+        private readonly ScriptFile _scriptFile;
+
+        internal EditorScriptFile(
+            ScriptFile scriptFile)
+        {
+            _scriptFile = scriptFile;
+            Uri = new Uri(scriptFile.DocumentUri);
+            Lines = _scriptFile.FileLines.AsReadOnly();
+        }
+
+        public Uri Uri { get; }
+
+        public IReadOnlyList<string> Lines { get; }
+
+        public string Content => _scriptFile.Contents;
+
+        public ScriptBlockAst Ast => _scriptFile.ScriptAst;
+
+        public IReadOnlyList<Token> Tokens => _scriptFile.ScriptTokens;
+    }
+
+    internal class WorkspaceService : IWorkspaceService
+    {
+        private readonly IEditorOperations _editorOperations;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unread private field '_editorOperations' or refactor the code to use its value.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4979332)